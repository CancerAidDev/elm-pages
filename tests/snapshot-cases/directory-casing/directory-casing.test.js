const { elmPagesCliFile, elmPagesUiFile } = require("../../../generator/src/elm-file-constants.js");
const generateRecords = require("../../../generator/src/generate-records.js");

test('casing is correct for directories', async () => {
    process.chdir(__dirname);
    const staticRoutes = await generateRecords();

    global.builtAt = new Date("Sun, 17 May 2020 16:53:22 GMT");
    expect(elmPagesUiFile(staticRoutes, [])).toMatchSnapshot();
});
