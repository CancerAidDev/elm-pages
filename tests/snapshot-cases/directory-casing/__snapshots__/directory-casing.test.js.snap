// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`casing is correct for directories 1`] = `
"port module Pages exposing (PathKey, allPages, allImages, internals, images, isValidRoute, pages, builtAt)

import Color exposing (Color)
import Pages.Internal
import Head
import Html exposing (Html)
import Json.Decode
import Json.Encode
import Pages.Platform
import Pages.Manifest exposing (DisplayMode, Orientation)
import Pages.Manifest.Category as Category exposing (Category)
import Url.Parser as Url exposing ((</>), s)
import Pages.ImagePath as ImagePath exposing (ImagePath)
import Pages.PagePath as PagePath exposing (PagePath)
import Pages.Directory as Directory exposing (Directory)
import Time


builtAt : Time.Posix
builtAt =
    Time.millisToPosix 1589734402000


type PathKey
    = PathKey


buildImage : List String -> ImagePath.Dimensions -> ImagePath PathKey
buildImage path dimensions =
    ImagePath.build PathKey (\\"images\\" :: path) dimensions


buildPage : List String -> PagePath PathKey
buildPage path =
    PagePath.build PathKey path


directoryWithIndex : List String -> Directory PathKey Directory.WithIndex
directoryWithIndex path =
    Directory.withIndex PathKey allPages path


directoryWithoutIndex : List String -> Directory PathKey Directory.WithoutIndex
directoryWithoutIndex path =
    Directory.withoutIndex PathKey allPages path


port toJsPort : Json.Encode.Value -> Cmd msg

port fromJsPort : (Json.Decode.Value -> msg) -> Sub msg


internals : Pages.Internal.Internal PathKey
internals =
    { applicationType = Pages.Internal.Browser
    , toJsPort = toJsPort
    , fromJsPort = fromJsPort identity
    , content = content
    , pathKey = PathKey
    }




allPages : List (PagePath PathKey)
allPages =
    [ (buildPage [ \\"blog-post\\", \\"post1\\" ])
    ]

pages =
    { blogPost =
        { post1 = (buildPage [ \\"blog-post\\", \\"post1\\" ])
        , directory = directoryWithoutIndex [\\"blog-post\\"]
        }
    , directory = directoryWithoutIndex []
    }

images =
    { directory = directoryWithoutIndex []
    }


allImages : List (ImagePath PathKey)
allImages =
    [
    ]


isValidRoute : String -> Result String ()
isValidRoute route =
    let
        validRoutes =
            List.map PagePath.toString allPages
    in
    if
        (route |> String.startsWith \\"http://\\")
            || (route |> String.startsWith \\"https://\\")
            || (route |> String.startsWith \\"#\\")
            || (validRoutes |> List.member route)
    then
        Ok ()

    else
        (\\"Valid routes:\\\\n\\"
            ++ String.join \\"\\\\n\\\\n\\" validRoutes
        )
            |> Err


content : List ( List String, { extension: String, frontMatter : String, body : Maybe String } )
content =
    [ 
    ]
"
`;
